libraries/chain/transaction_context.cpp

```diff
@@ -647,6 +647,7 @@ namespace eosio { namespace chain {
         if (auto dm_logger = control.get_deep_mind_logger()) {
             event_id = STORAGE_EVENT_ID("${id}", ("id", gto.id));
 
+            auto packed_signed_trx = fc::raw::pack(trx);
```

Comment:
`packed_signed_trx` materially different from `packed_trx` ?  This looks like an unnecessary  "round trip" with line 626 above

Reply:
I'll double check, I remember some had subtle differences when packing not giving the exact same output format, specially related to signature. One was giving an output with signature the other not, but I don't remember if this one was problematic or not.

Reply:
So, it appears `packed_trx` would

- potentially have signatures (relevant if it weren't a deferred transaction)
- be potentially compressed (relevant if it weren't a deferred transaction)
- represent the (un)?compressed transaction as a blob of data and contain some envelope information about the blob's format

by re-packing it it should always be uncompressed and only the transaction BUT i don't think it will have signatures.  

`packed_transaction::get_transaction() const` returns a `transaction` not a `signed_transaction`.  Also, thisis a deferred transaction which has no signatures.  Please rename the variable to something like `serialized_trx` so that future readers do not expect this blob to contain signatures.

Reply:
In version 2.x of deep mind branch, this was outputting signature (for delayed pushed transaction which is the case here). You are right that is not a `signed_trx` anymore here, it's a plain transaction without the signature. 

Now, I checked and using `fc::to_hex(packed_trx.get_packed_transaction().data(),  packed_trx.get_packed_transaction().size())` does not print signature neither. That would have been a correct change since we could have uncompress the transaction if it was compressed.

Now, I think my best way of handling it here is to also output the signatures directly, hoping they are not pruned which seems the case here.

This does not compile yet because `error: no member named 'visit' in 'fc::reflector<const std::__1::vector<fc::crypto::signature, std::__1::allocator<fc::crypto::signature> > *>'` when used in a mutable variable object for printing like this:

```
            fc_dlog(*dm_logger, "DTRX_OP PUSH_CREATE ${action_id} ${sender} ${sender_id} ${payer} ${published} ${delay} ${expiration} ${trx_id} ${trx} ${signatures}",
               ("action_id", get_action_id())
...
               ("signatures", packed_trx.get_signatures())
            );
```

I still have some holes in my understanding of the bigger serialization format around fc and EOSIO codebase, but I don't see yet why it complains like this. The `fc::crypto::signature` has specialized `to_variant` and `from_variant` overload as well as having a dedicated `FC_REFLECT(fc::crypto::signature, (_storage) )` definition.

So maybe you could give me some hint here to find why it's not working.

Reply:
You have two options here:

1. maintain the course with your new serialization, you should dereference the pointer so that the `mutable_variant_object` receives a `const std::vector<signature_type>&` instead of a pointer.

2. go back to the original and change it to:
```
auto packed_signed_trx = fc::raw::pack(packed_trx.to_packed_transaction_v0().get_signed_transaction());
```
which should give you what you were seeing in the 2.x version

Reply:
Thanks for the tips.

I went with the original output and the extra hop of turning it into a v0 packed transaction. It was easier to go that route, specially since I would also needed to add `context_free_data`.